### ISSUE TITLE
[ALM Accelerator - COE CLI]: Nothing happens during installation.

### ISSUE THREAD

**Issue by TBag** [microsoft/coe-starter-kit#1856]
### Describe the issue

@mikefactorial Have you seen this one before?

First, I created the following install file using the COE CLI.

{
  "log": [
    "debug"
  ],
  "components": [
    "all"
  ],
  "aad": "ALMAcceleratorServicePrincipal",
  "group": "ALMAcceleratorForMakers",
  "devOpsOrganization": "https://dev.azure.com/CRM034681",
  "project": "alm-sandbox",
  "repository": "alm-sandbox",
  "pipelineRepository": "pipelines",
  "environments": "Contoso (default)",
  "settings": {
    "installEnvironments": [
      "validation",
      "test",
      "prod"
    ],
    "validation": "CRM034681-validation",
    "test": "CRM034681-test",
    "prod": "CRM034681-prod",
    "createSecret": "true",
    "region": [
      "NAM"
    ]
  },
  "importMethod": "api",
  "endpoint": "prod",
  "$schema": "./alm.schema.json"
}

Then, I tried to install the AA4PP with the COE CLI and the install file.  Nothing happens.

![image](https://user-images.githubusercontent.com/1407310/151249069-353f33f8-73a1-4461-8743-f32b2bfbabb7.png)

Logs don't provide any clues, even when set to verbose and debug.  Here's the logs.

{"level":"info","message":"Generate Install start","timestamp":"2022-01-26 03:34:36 PM -0500"}
{"level":"info","message":"Generate Install end","timestamp":"2022-01-26 03:35:55 PM -0500"}
{"level":"info","message":"Install start","timestamp":"2022-01-26 04:08:42 PM -0500"}
{"level":"info","message":"Loading configuration","timestamp":"2022-01-26 04:08:42 PM -0500"}
{"level":"info","message":"Starting install","timestamp":"2022-01-26 04:08:42 PM -0500"}
{"level":"info","message":"Install start","timestamp":"2022-01-26 04:09:53 PM -0500"}
{"level":"info","message":"Loading configuration","timestamp":"2022-01-26 04:09:53 PM -0500"}
{"level":"info","message":"Starting install","timestamp":"2022-01-26 04:09:53 PM -0500"}
{"level":"info","message":"Install start","timestamp":"2022-01-26 04:11:23 PM -0500"}
{"level":"info","message":"Loading configuration","timestamp":"2022-01-26 04:11:23 PM -0500"}
{"level":"info","message":"Starting install","timestamp":"2022-01-26 04:11:23 PM -0500"}
{"level":"info","message":"Install start","timestamp":"2022-01-26 04:12:03 PM -0500"}
{"level":"info","message":"Loading configuration","timestamp":"2022-01-26 04:12:03 PM -0500"}
{"level":"info","message":"Install start","timestamp":"2022-01-26 04:24:45 PM -0500"}
{"level":"info","message":"Loading configuration","timestamp":"2022-01-26 04:24:45 PM -0500"}

Any ideas?

### Expected Behavior

The AA4PP is installed.

### What component are you experiencing the issue with?

ALM Accelerator Canvas App

### What solution version are you using?

Preview

### Steps To Reproduce

Follow the steps described above.

**Comment by davidyack** [comment]
I had issues with environment names (including dev,test and prod)  really wanting URLs - script and docs were not clear on that

**Comment by TBag** [comment]
Thanks @mikefactorial , the files are in the same directory.  

@davidyack Thanks for chiming in.  OK, I will try the URL change.  Did you include https:// in the URLs?

**Comment by mikefactorial** [comment]
One of the first things the install does is tries to get a token for Azure using az login. It's possible that the interactive browser window for that popped up in the background or got suppressed.

**Comment by mikefactorial** [comment]
@Grant-Archibald-MS any thoughts?

**Comment by Grant-Archibald-MS** [comment]
> I had issues with environment names (including dev,test and prod) really wanting URLs - script and docs were not clear on that

@davidyack reedback make sense. Will update docs and help pages to make this clearer

**Comment by davidyack** [comment]
@TBag  yeah I used fully qualified e.g, https://dev-l17.crm.dynamics.com

but it does look like you are stuck before that as Mike said...perhaps try just running the az login and make sure that works

**Comment by Grant-Archibald-MS** [comment]
@mikefactorial looking at logs above I think it is getting the access tokens to the environments that is the problem

I will add a new branch with additional audit logs so we can verify
Another change would be to update the environment process to lookup name and convert to url. This would require a bit more refactoring to make this change

**Comment by Grant-Archibald-MS** [comment]
@TBag I have created a new branch with some additional logging that could help diagnose the issue.

Also looking at config file are you specifying a value url values for **validation**, **test** and **prod**?

```pwsh
git checkout main
git pull
git checkout grarchib-coe-cli-1856
cd coe-cli
npm run build
coe alm install -f AA4PP.json
```

to switch back to main branch

```
git checkout main
cd coe-cli
npm build
```

**Comment by TBag** [comment]
@mikefactorial @Grant-Archibald-MS @davidyack 

Hey fellas,

Thanks for all the help here.  I took a pretty thorough approach to ensure we found the culprit.

First, I cloned Grant's branch and tried the install file with the same settings you see in this Issue.  I got the same results, which were, nothing happens.  Here's the output from the logs, nothing helpful.

{"level":"info","message":"Install start","timestamp":"2022-01-27 05:20:49 PM -0500"}
{"level":"info","message":"Loading configuration","timestamp":"2022-01-27 05:20:49 PM -0500"}
{"level":"info","message":"Install started","timestamp":"2022-01-27 05:20:49 PM -0500"}

The additional logging you output to the console showed up and it confirms your theory about the access tokens.

![image](https://user-images.githubusercontent.com/1407310/151584260-47c75a58-7830-4c0d-be9e-22f5e6fc1bcf.png)

Then, I updated the settings file to use the full URLs for ALL the environments and the script ran all the way to the end!  Time to start testing.  :)

I appreciate you guys, thank you again.

Todd
